src/main/java/ua/utility/kfsdbupgrade/ForeignKeyReference.java

```diff
@@ -105,7 +105,16 @@ private String buildIndexName(int indx) {
 		StringBuilder retval = new StringBuilder(128);
 		retval.append(schemaName.toUpperCase());
 		retval.append(".");
-		retval.append(indexNameTemplate.replace("[table-name]", tableName).replace("{index}", "" + indx));
+		// Oracle before v 12.2 only allows identifiers upto 30 characters in length, so shorten index names.
+		// Instead of using the table name as the index name stub use the table name with
+		// trailing "_T" (underscore T) removed and all "_"'s (underscores) removed as the index name stub.
+		// Thus the index name will be shorter but still correlate with the unmolested table name.
+		String shortIndexNameStub = tableName;
+		if (tableName.lastIndexOf("_T") == (tableName.length() - 2)) {
```

Comment:
I'd suggest the following for the conditional:
`foo.toLowerCase().endsWith("_t")`

Reads a little easier, and also will be lettercase agnostic.